### PR TITLE
refactor(app): simplify UpdateAppModal to remove a click

### LINKED ISSUE #6682: refactor(app): Show release notes on first screen of app update modal
## Overview

As a user, I'd like to view the update release notes immediately upon being notified of an app update.

## Acceptance criteria

- [ ] When an app update notification pops up, the release notes are shown
- [ ] Clicking the "View Available Update" button in the app info page shows the release notes

## Implementation details

- In the `<AppUpdateModal>` wizard, remove the first page

### PR DESCRIPTION
**Pull request by mcous**
## overview

This PR simplifies the `<UpdateAppModal>` to:

- Remove the initial screen before the release notes, because it's unhelpful
- Prepare the component to move out of routes and into alert logic via #6683

Closes #6682. See ticket for acceptance criteria and details.

## changelog

- refactor(app): simplify UpdateAppModal to remove a click 

# review requests

Testing this one is a little funny, because it's pretty difficult to exercise the update mechanisms without a production, signed, app. For some simple smoke testing, you can trick your dev app into thinking there's an update by fiddling with `package.json::version` in both `app/package.json` and `app-shell/package.json`. From there:

- [ ] App update notification should launch when you open the app
- [ ] Notification takes you straight to release notes
- [ ] You should be able to click "Not Now"
- [ ] You should be able to click "View Available Update" in More > App, which opens the release notes back up
- [ ] You can still click "Not Now"

On the dev app on macOS, clicking "Download" on the update notification will always error out. You'll probably see:

- [ ] A spinner for a second
- [ ] Then an error message

Showing a spinner properly and an error message properly are also part of the smoke test, though, so it's worth doing. I can build a mock production app locally, so I can smoke test the success flow. All these flows now have unit tests, which they did not have before.

# risk assessment

Low; UI changes to the up update flow, but the changes to not affect any of the update mechanisms themselves. PR includes unit tests to ensure UI is still dispatching the proper update actions in the proper order.

### CODE REVIEW

**Review by sfoster1 — APPROVED** [review]
Smoke test passed on my Linux dev setup, styling looks good.

**Review by shlokamin — COMMENTED** [review]
Code looks good to me — there's some leftover `todos` in `AppInfoCard.test.js` that seem to be wanting to test some important functionality though.

File: app/src/components/AppSettings/__tests__/AppInfoCard.test.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    const wrapper = render()
+    const versionValue = wrapper.find(LabeledValue)
+
+    expect(versionValue.prop('label')).toBe('Software Version')
+    expect(versionValue.prop('value')).toBe(CURRENT_VERSION)
+  })
+
+  it.todo('should have a SecondaryBtn that is disabled if no update available')
+
+  it.todo('clicking the SecondaryBtn should render an <UpdateAppModal>')
```
[the remark this answers is not in the document]
Comment (mcous) [reply app/src/components/AppSettings/__tests__/AppInfoCard.test.js:48]: Nope, that functionality will be a part of #6683, which will switch the `<UpdateAppModal>` from being displayed according the routes to being displayed according to state. Once that state exists, this test will have something to actually check

File: app/src/shell/update.js [hunk 2]
```diff
@@ -25,13 +25,15 @@ export function setShellUpdateSeen(): ShellUpdateAction {
...
 }
 
-type StringSelector = State => ?string
-
 export function getShellUpdateState(state: State): ShellUpdateState {
   return state.shell.update
 }
 
-export const getAvailableShellUpdate: StringSelector = createSelector(
+export function getShellUpdateInfo(state: State): UpdateInfo | null {
```
Comment (mcous) [inline app/src/shell/update.js:32]: Oops, this selector isn't actually needed. Should be removed

**Comment by mcous** [comment]
Smoke tested on my machine and everything checks out. Here's what I tested:

- [x] After the update is downloaded, the update instructions are shown
    - (the "update instructions" say to update your robot and all other apps you use with your robot)
- [x] The "restart app" button on the download success screen still works properly
- [x] The logic in robot update flow to shunt you to the app update first if one is available still works

File: app/src/components/AppSettings/UpdateAppModal.js [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+        <Text>
+          You should update the Opentrons App on <strong>all computers</strong>{' '}
+          that you use with your OT-2.
+        </Text>
+      </li>
+    </Box>
+  </>
+)
+
+const SPINNER = (
```
Comment (b-cooper) [inline app/src/components/AppSettings/UpdateAppModal.js:76]: nit: we normally name component instances using camelCase, is there a good reason to use CAPITAL_CASE here?

**Review by b-cooper — APPROVED** [review]
🏞️ looks good

File: app/src/components/AppSettings/__tests__/AppInfoCard.test.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (shlokamin) [reply app/src/components/AppSettings/__tests__/AppInfoCard.test.js:48]: Gocha! Not introduced in this PR but might be worth checking that update button (`SecondaryBtn`) has the correct link too

File: app/src/components/AppSettings/UpdateAppModal.js (hunk 3 above)
Comment (mcous) [reply app/src/components/AppSettings/UpdateAppModal.js:76]: I was going off our convention to name constant instance variables at the module scope (as opposed to in the scope of some function) in all caps.

Our codebase doesn't seem to offer much of a consensus on module-scoped const React components. From some very quick regex searching I found around 7 ALL_CAPS const component instances (most admittedly written by me) and one camelCase. I don't much care, and I can see how all caps could feel awkward, so happy to have a discussion about convention here

**Pull request merged**